src/test/java/org/mule/tools/apikit/AbstractScaffolderTestCase.java

```diff
@@ -40,6 +48,63 @@ public void after() {
     System.clearProperty(TestUtils.PARSER_V2_PROPERTY);
   }
 
+  protected static void verifySuccessfulScaffolding(ScaffoldingResult result, String... expectedArrayPath)
+      throws IOException, SAXException {
+    List<String> expectedPaths = Arrays.asList(expectedArrayPath);
+    assertTrue(result.isSuccess());
+    assertEquals(expectedPaths.size(), result.getGeneratedConfigs().size());
+    for (String expectedPath : expectedArrayPath) {
+      String expected = APIKitTools.readContents(getResourceAsStream(expectedPath));
+      String configFileName = extractName(expectedPath);
+      String generated = retrieveGeneratedFile(result, configFileName);
+      XMLUnit.setIgnoreWhitespace(true);
+      Diff diff = XMLUnit.compareXML(expected, generated);
+      assertTrue(diff.identical());
+    }
+  }
+
+  protected static String retrieveGeneratedFile(ScaffoldingResult secondScaffoldingResult, String fileName)
+      throws IOException {
+    InputStream api = secondScaffoldingResult.getGeneratedConfigs().stream().filter(config -> config.getName().contains(fileName))
+        .findFirst().get().getContent();
+    return APIKitTools.readContents(api);
+  }
+
+  protected static MuleConfig createConfig(String path) throws Exception {
+    InputStream resourceAsStream = getResourceAsStream(path);
+    MuleConfig muleConfig = MuleConfigBuilder.fromStream(resourceAsStream);;
+    muleConfig.setName(extractName(path));
+    return muleConfig;
+  }
+
+  protected static String extractName(String path) {
+    String name = "";
+    String[] pathParts = path.split("/");
+    name = pathParts[pathParts.length - 1];
+    return name;
+  }
+
+  // Check if this method could be replaced with already existing scaffoldAPI().
```

Comment:
I took a quick look and it seems possible to merge this with the `scaffoldApi`, you just need refactor the muleConfigs generation in there.